steps/rabbit/session.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbit
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Telefonica/golium"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/streadway/amqp"
+	"github.com/tidwall/sjson"
+)
+
+// MsgWithCorr contains the information of a message with a corresponding correlator.
+type MsgWithCorr struct {
+	Message    string
+	Correlator string
+}
+
+// Session contains the information of a rabbit session.
+type Session struct {
+	Connection *amqp.Connection
+	// Messages received from the publish/subscribe channel
+	Messages []MsgWithCorr
+	// SessionCorrelator is used to correlate the messages for a specific session
+	SessionCorrelator string
```

Comment:
I feel that using a Message+Correlator (MsgWithCorr) is a bit artificial and very specific of what we want to implement.
Note that we are misusing the messageId and correlationId. For example, the `correlationId` is used to build an RPC pattern in order to match a request message with the corresponding response message. But this is not the case.
I guess that it would be better to create a step to add headers to the rabbit message, and one of this headers could be the correlator.